modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/upload/DDMFormUploadFileEntryHandler.java

```diff
@@ -88,7 +99,8 @@ protected FileEntry addFileEntry(
 			String fileName, String mimeType, ThemeDisplay themeDisplay)
 		throws PortalException {
 
-		if (!DDMFormInstancePermission.contains(
+		if ((formInstanceId != 0) &&
```

Comment:
@linolaoj, did you add this other check because you see no reason (no way) to check permissions for a not persisted Form, did I understand you correctly?

Did you do it due to some bug?

Reply:
@linolaoj Shouldn't we keep checking permission to add document?

```
ModelResourcePermissionUtil.check(
	_folderModelResourcePermission, themeDisplay.getPermissionChecker(),
	themeDisplay.getScopeGroupId(), folderId, ActionKeys.ADD_DOCUMENT);
```

This code is in the `DDMUserPersonalFolderUploadFileEntryHandler` class and was removed in this PR.

Reply:
Hi @natocesarrego , @carolmariaabb , on PTR-2604 at this [comment](https://issues.liferay.com/browse/PTR-2604?focusedCommentId=2515077&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2515077) it says that DLAppService should not be used, so I removed all the code related to Document Library. That is why the folder permission was also removed. When uploading the file the `formInstanceId` is null, I believe there is a better way to check this permission, but I could not find it.